libs/core/algorithms/include/hpx/parallel/algorithms/detail/find.hpp

```diff
@@ -256,4 +256,221 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
+            std::forward<Pred>(op), std::forward<Proj1>(proj1),
+            std::forward<Proj2>(proj2));
+    }
+
+    template <typename ExPolicy, typename Iter1, typename Iter2, typename Token,
+        typename Pred, typename Proj1, typename Proj2>
+    HPX_HOST_DEVICE HPX_FORCEINLINE Iter1 sequential_find_end(Iter1 it,
+        Iter2 first2, std::size_t base_idx, std::size_t part_size,
+        std::size_t diff, Token& tok, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+    {
+        return sequential_find_end_t<ExPolicy>{}(it, first2, base_idx,
+            part_size, diff, tok, std::forward<Pred>(op),
+            std::forward<Proj1>(proj1), std::forward<Proj2>(proj2));
+    }
+#endif
+
+    template <typename ExPolicy>
+    struct sequential_find_first_of_t
+      : hpx::functional::detail::tag_fallback<
+            sequential_find_first_of_t<ExPolicy>>
+    {
+        template <typename InIter1, typename InIter2, typename Pred,
+            typename Proj1, typename Proj2>
+        friend inline constexpr InIter1 tag_fallback_invoke(
+            sequential_find_first_of_t<ExPolicy>, InIter1 first, InIter1 last,
+            InIter2 s_first, InIter2 s_last, Pred&& op, Proj1&& proj1,
+            Proj2&& proj2)
+        {
+            if (first == last)
+                return last;
+
+            for (/* */; first != last; ++first)
+            {
+                for (InIter2 iter = s_first; iter != s_last; ++iter)
+                {
+                    if (hpx::util::invoke(
+                            util::compare_projected<Pred, Proj1, Proj2>(
+                                std::forward<Pred>(op),
+                                std::forward<Proj1>(proj1),
+                                std::forward<Proj2>(proj2)),
+                            *first, *iter))
+                    {
+                        return first;
+                    }
+                }
+            }
+            return last;
+        }
+
+        template <typename FwdIter, typename FwdIter2, typename Token,
+            typename Pred, typename Proj1, typename Proj2>
+        friend inline constexpr void tag_fallback_invoke(
+            sequential_find_first_of_t<ExPolicy>, FwdIter it, FwdIter2 s_first,
+            FwdIter2 s_last, std::size_t base_idx, std::size_t part_size,
+            Token& tok, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+        {
+            util::loop_idx_n<ExPolicy>(base_idx, it, part_size, tok,
+                [&tok, &s_first, &s_last, &op, &proj1, &proj2](
+                    auto v, std::size_t i) -> void {
+                    for (FwdIter2 iter = s_first; iter != s_last; ++iter)
+                    {
+                        if (hpx::util::invoke(
+                                util::compare_projected<Pred, Proj1, Proj2>(
+                                    std::forward<Pred>(op),
+                                    std::forward<Proj1>(proj1),
+                                    std::forward<Proj2>(proj2)),
```

Comment:
Same here for now forwarding?

Reply:
The `util::compare_projected<>` can be constructed outside of the loops, then the forwarding would be correct.